queries/places.py

```diff
@@ -0,0 +1,352 @@
+
+QPlacesIsOpen →
+    "er" "opið" QPlacesPrepAndSubject QPlNow?
+    | "er" QPlacesSubjectNf QPlOpen QPlNow?
+
+QPlacesIsClosed →
+    "er" "lokað" QPlacesPrepAndSubject QPlNow?
+    | "er" QPlacesSubjectNf QPlClosed QPlNow?
+
+QPlacesAddress →
+    "hvert" "er" "heimilisfangið" QPlacesPrepAndSubject
+    | "hvað" "er" "heimilisfangið" QPlacesPrepAndSubject
+    | "hvert" "er" "heimilisfang" QPlacesSubjectEf
+    | "hvað" "er" "heimilisfang" QPlacesSubjectEf
+    | "hvar" "er" QPlacesSubjectNf "til" "húsa"
+    | "hvar" "er" QPlacesSubjectNf "staðsett" 
+    | "hvar" "er" QPlacesSubjectNf "staðsettur"
+    | QPlacesPreposition "hvaða" "götu" "er" QPlacesSubjectNf
+    | QPlacesPreposition "hvaða" "stræti" "er" QPlacesSubjectNf
+
+QPlacesPrepAndSubject →
+    QPlacesPrepWithÞgf QPlacesSubjectÞgf
+    | QPlacesPrepWithÞf QPlacesSubjectÞf
+
+QPlacesSubjectNf →
+    Nl_nf
+
+QPlacesSubjectÞf →
+    Nl_þf
+
+QPlacesSubjectÞgf →
+    Nl_þgf
+
+QPlacesSubjectEf →
+    Nl_ef
+
+QPlacesPreposition →
+    QPlacesPrepWithÞf | QPlacesPrepWithÞgf
+
+QPlacesPrepWithÞgf →
+    "á" | "í" | "hjá"
+
+QPlacesPrepWithÞf →
+    "við" | "fyrir"
+
+QPlOpen →
+    "opið" | "opin" | "opinn"
+
+QPlClosed →
+    "lokað" | "lokuð" | "lokaður"
+
+QPlNow →
+    "núna" | "í" "augnablikinu" | "eins" "og" "stendur" | "nú"
+
+QPlToday →
+    "núna"? "í" "dag" | "núna"? "í" "kvöld"
+
+$score(+35) QPlacesQuery
+
+"""
+
+
+_PLACENAME_MAP = {}
+
+
+def _fix_placename(pn):
+    p = pn.capitalize()
+    return _PLACENAME_MAP.get(p, p)
+
+
+def QPlacesQuery(node, params, result):
+    result["qtype"] = _PLACES_QTYPE
+
+
+def QPlacesOpeningHours(node, params, result):
+    result["qkey"] = "OpeningHours"
+
+
+def QPlacesIsOpen(node, params, result):
+    result["qkey"] = "IsOpen"
+
+
+def QPlacesIsClosed(node, params, result):
+    result["qkey"] = "IsClosed"
+
+
+def QPlacesAddress(node, params, result):
+    result["qkey"] = "PlaceAddress"
+
+
+def QPlacesSubjectNf(node, params, result):
+    result["subject_nom"] = _fix_placename(result._nominative)
+
+
+QPlacesSubjectEf = QPlacesSubjectÞgf = QPlacesSubjectÞf = QPlacesSubjectNf
+
+
+_PLACES_API_ERRMSG = "Ekki tókst að fletta upp viðkomandi stað"
+_NOT_ICELAND_ERRMSG = "Enginn staður með þetta heiti fannst á Íslandi"
+
+
+def _parse_coords(place):
+    try:
+        lat = float(place["geometry"]["location"]["lat"])
+        lng = float(place["geometry"]["location"]["lng"])
+        return (lat, lng)
+    except Exception as e:
+        logging.warning(
+            "Unable to parse place coords for place {0}: {1}".format(place, e)
+        )
+    return None
+
+
+def answ_address(placename, loc, qtype):
+    # Look up placename in places API
+    res = query_places_api(
+        placename, userloc=loc, fields="formatted_address,name,geometry"
+    )
+
+    if res["status"] != "OK" or "candidates" not in res or not res["candidates"]:
+        return gen_answer(_PLACES_API_ERRMSG)
+
+    # Use top result
```

Comment:
Might want to use top result that is located in Iceland!

```
for place in res["candidates"]:
    coords = _parse_coords(place)
    if not coords or not in_iceland(coords):
        continue
    break
else:
    return gen_answer(_NOT_ICELAND_ERRMSG)

```

Reply:
Haha, good call! Don't know what I was thinking...